### PR TITLE
[Facilities] Add facility markers for each location

### PR DESCRIPTION
**Pull request by linyc0817**
# Description
Clicking on a cluster will display facility markers in that location

# Changes
1. Add facility json files for each location(cluster).
2. Check data name in the  cluster to determine which facility data to show

# Notes
Facility markers will show only when its cluster is open

# Checklist

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] Any changes to strings have been published to our translation tool
- [x] I conducted basic QA to assure all features are working
- [x] I requested code review from other team members

# Resolved Issues
#11

### CODE REVIEW

File: src/constants/facility.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import oldGymData from "../assets/data/markers/facilities/oldgym.json";
+import libraryData from "../assets/data/markers/facilities/library.json";
+
+function getLocations(clusterName: string): Facilities {
+  let facilities: Facilities = {};
+  // let dataFromJson: DataFromJson = [];
+
+  switch (true) {
+    case /^小木屋&校計中/.test(clusterName):
+      facilities = shineMoodCCData;
```
Comment (y-c-wang) [inline src/constants/facility.ts:17]:
```suggestion
      return shineMoodCCData;
```

By doing so we do not need to maintain `facilities`.

File: src/components/Map/Markers/FacilityMarkers/index.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    if (cluster) {
+      return getLocations(cluster.data.name);
+    }
+    return {};
+  }, [cluster]);
+
+  React.useEffect(() => {
+    if (isCurrentSearchParamsCluster)
+      markers.facility.setFacilities(facilities);
+    else markers.facility.clear();
```
Comment (y-c-wang) [inline src/components/Map/Markers/FacilityMarkers/index.tsx:33]:
```suggestion
    if (isCurrentSearchParamsCluster) {
      markers.facility.setFacilities(facilities);
    } else {
      markers.facility.clear();
    }
```

File: src/components/Map/Markers/index.tsx [hunk 3]
```diff
@@ -3,12 +3,14 @@ import React from "react";
 import ClusterMarkers from "./ClusterMarkers";
 import PoiMarkers from "./PoiMarkers";
 import UserMarker from "./UserMarker";
+import FacilityMarker from "./FacilityMarkers";
```
Comment (y-c-wang) [inline src/components/Map/Markers/index.tsx:6]:
```suggestion
import ClusterMarkers from "./ClusterMarkers";
import FacilityMarker from "./FacilityMarkers";
import PoiMarkers from "./PoiMarkers";
import UserMarker from "./UserMarker";
```

File: src/components/Map/Markers/index.tsx [hunk 4]
```diff
@@ -3,12 +3,14 @@ import React from "react";
...
 
 const Markers: React.FC = () => {
   return (
     <>
       <ClusterMarkers />
       <PoiMarkers />
+      <FacilityMarker />
       <UserMarker />
```
Comment (y-c-wang) [inline src/components/Map/Markers/index.tsx:14]:
```suggestion
      <ClusterMarkers />
      <FacilityMarker />
      <PoiMarkers />
      <UserMarker />
```

File: src/constants/facility.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import { Facilities } from "../models/facility";
+import shineMoodCCData from "../assets/data/markers/facilities/shinemood-cc.json"; // 小木屋&校計中
```
Comment (y-c-wang) [inline src/constants/facility.ts:2]:
```suggestion
import shineMoodCCData from "../assets/data/markers/facilities/shinemood-cc.json";
```

File: src/utils/googleMaps/markers/index.ts [hunk 6]
```diff
@@ -1,3 +1,4 @@
 export * as user from "./user";
 export * as poi from "./poi";
 export * as cluster from "./cluster";
+export * as facility from "./facility";
```
Comment (y-c-wang) [inline src/utils/googleMaps/markers/index.ts:4]:
```suggestion
export * as cluster from "./cluster";
export * as facility from "./facility";
export * as poi from "./poi";
export * as user from "./user";
```

File: src/constants/facility.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+import caCafe1Data from "../assets/data/markers/facilities/ca-cafe1.json";
+import parkingLotCafe2Data from "../assets/data/markers/facilities/parkinglot-cafe2.json";
+import swimMultiGymData from "../assets/data/markers/facilities/swim-multigym.json";
+import outdoorFieldData from "../assets/data/markers/facilities/outdoor-field.json";
+import oldGymData from "../assets/data/markers/facilities/oldgym.json";
+import libraryData from "../assets/data/markers/facilities/library.json";
+
+function getLocations(clusterName: string): Facilities {
+  let facilities: Facilities = {};
+  // let dataFromJson: DataFromJson = [];
```
[the remark this answers is not in the document]
Comment (linyc0817) [reply src/constants/facility.ts:13]: done

File: src/constants/facility.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+  switch (true) {
+    case /^小木屋&校計中/.test(clusterName):
```
[the remark this answers is not in the document]
Comment (linyc0817) [reply src/constants/facility.ts:16]: 我不是很理解這個 comment , 你是指 cluster 裡面多一個 para 專門存給 facility markers 判斷地點嗎？

File: src/components/Map/Markers/FacilityMarkers/index.tsx (hunk 2 above)
Comment (linyc0817) [reply src/components/Map/Markers/FacilityMarkers/index.tsx:33]: done

File: src/components/Map/Markers/index.tsx (hunk 3 above)
Comment (linyc0817) [reply src/components/Map/Markers/index.tsx:6]: done

File: src/components/Map/Markers/index.tsx (hunk 4 above)
Comment (linyc0817) [reply src/components/Map/Markers/index.tsx:14]: done

File: src/constants/facility.ts (hunk 5 above)
Comment (linyc0817) [reply src/constants/facility.ts:2]: done

File: src/utils/googleMaps/markers/index.ts (hunk 6 above)
Comment (linyc0817) [reply src/utils/googleMaps/markers/index.ts:4]: done

File: src/constants/facility.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (y-c-wang) [reply src/constants/facility.ts:16]: Exactly, we may modify the data of cluster stored in firebase:)

[the remark this answers is not in the document]
Comment (linyc0817) [reply src/constants/facility.ts:13]: Postpone, maybe we can do this in the later version (Refactor)

**Review by y-c-wang — APPROVED** [review]

**Pull request merged**